x-pack/plugins/apm/public/components/shared/DatePicker/index.tsx

```diff
@@ -73,6 +74,7 @@ export function DatePicker() {
   }
 
   function onTimeChange({ start, end }: { start: string; end: string }) {
+    plugins.data.query.timefilter.timefilter.setTime({ from: start, to: end });
```

Comment:
is this function also called when the URL changes? What happens when you select a different time range and press the back button?

Reply:
I think you are right but didn't get around to verify whether it was a problem it since I refactored it anyway. LMK if you still think it could be an issue